src/Symfony/Component/Translation/Extractor/PhpAstExtractor.php

```diff
@@ -46,13 +46,22 @@ public function extract(iterable|string $resource, MessageCatalogue $catalogue):
     {
         foreach ($this->extractFiles($resource) as $file) {
             $traverser = new NodeTraverser();
+
+            $nodes = $this->parser->parse(file_get_contents($file));
+
+            // First run is needed to resolve namespaces in class methods/constants.
```

Comment:
Could we visit this as a single run instead of 2 ?

Reply:
I didn't think so.

With The following code
```
            $nameResolver = new NodeVisitor\NameResolver();
            $traverser->addVisitor($nameResolver);

            /** @var AbstractVisitor&NodeVisitor $visitor */
            foreach ($this->visitors as $visitor) {
                $visitor->initialize($catalogue, $file, $this->prefix);
                $traverser->addVisitor($visitor);
            }

            $traverser->traverse($nodes);
```
tests are failing.

but starting thinking about it I would say it's because in the call
```
trans('message', [], App::DOMAIN);
```
We enter the `trans` NODE before the `APP::DOMAIN` so when we're resolving the name is too late.

So I think I found a solution => The TransVisitor (and other symfony visitors) should process translation when LEAVING a node, and not entering in it.
https://github.com/symfony/symfony/pull/53624/commits/8498b80f21e3e001647afc8f1ec59aaa23a42bb3